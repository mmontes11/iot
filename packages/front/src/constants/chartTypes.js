export const BARCHART = "BARCHART";
export const LINECHART = "LINECHART";
export const REALTIME = "REALTIME";
