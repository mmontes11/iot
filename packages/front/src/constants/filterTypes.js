export const THING_FILTER_TYPE = "thing";
export const DATE_FILTER_TYPE = "date";
export const FILTER_TYPES = [THING_FILTER_TYPE, DATE_FILTER_TYPE];
