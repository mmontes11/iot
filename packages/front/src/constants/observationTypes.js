export const EVENT_TYPE = "event";
export const MEASUREMENT_TYPE = "measurement";
export const OBSERVATION_TYPES = [EVENT_TYPE, MEASUREMENT_TYPE];
