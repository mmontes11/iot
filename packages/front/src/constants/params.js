export const TYPE = "TYPE";
export const OBSERVATION = "OBSERVATION";
export const GROUPBY = "GROUPBY";
export const THING = "THING";
