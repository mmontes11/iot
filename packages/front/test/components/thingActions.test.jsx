import React from "react";
import { shallow } from "enzyme";
import ThingActions from "components/thingActions";
import { thing } from "../constants";

describe("components/thingActions", () => {
  const onStatsClick = jest.fn();
  const onDataClick = jest.fn();
  const wrapper = shallow(<ThingActions thing={thing} onStatsClick={onStatsClick} onDataClick={onDataClick} />);
  it("renders", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("simulates clicks on actions", () => {
    wrapper.find("#google-maps-link").simulate("click");
    wrapper.find("#event-stats-button").simulate("click");
    wrapper.find("#measurement-stats-button").simulate("click");
    expect(onStatsClick).toHaveBeenCalled();
  });
});
