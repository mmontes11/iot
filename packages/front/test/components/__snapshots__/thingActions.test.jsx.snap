// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/thingActions renders 1`] = `
<p
  className="buttons is-horizontal-center"
>
  <a
    className="button"
    href="https://www.google.es/maps"
    id="google-maps-link"
    rel="noopener noreferrer"
    target="_blank"
  >
    <span
      className="icon"
    >
      <i
        className="fas fa-map-marked"
      />
    </span>
    <FormattedMessage
      id="Google maps"
      values={Object {}}
    />
  </a>
  <button
    className="button"
    id="event-stats-button"
    onClick={[Function]}
  >
    <span
      className="icon"
    >
      <i
        className="fas fa-chart-bar"
      />
    </span>
    <FormattedMessage
      id="Event stats"
      values={Object {}}
    />
  </button>
  <button
    className="button"
    id="measurement-stats-button"
    onClick={[Function]}
  >
    <span
      className="icon"
    >
      <i
        className="fas fa-chart-bar"
      />
    </span>
    <FormattedMessage
      id="Measurement stats"
      values={Object {}}
    />
  </button>
</p>
`;
